src/schema/associations.ts

```diff
@@ -123,33 +146,37 @@ export async function buildAssociations(
         rule.target.suffix,
         rule.target?.entities ?? [],
       ).next().value
-      if (Array.isArray(file)) {
-        file = file[0]
-      }
-    } catch (error) {
-      if (
-        error && typeof error === 'object' && 'code' in error &&
-        error.code === 'MULTIPLE_INHERITABLE_FILES'
-      ) {
-        // @ts-expect-error
+    } catch (error: any) {
+      if (error?.code === 'MULTIPLE_INHERITABLE_FILES') {
         context.dataset.issues.add(error)
-        break
+        continue
       } else {
         throw error
       }
     }
 
-    if (file) {
-      // @ts-expect-error
-      const load = associationLookup[key] ?? defaultAssociation
-      // @ts-expect-error
-      associations[key] = await load(file, { maxRows: context.dataset.options?.maxRows }).catch(
-        (error: any) => {
-          if (error.code) {
-            context.dataset.issues.add({ ...error, location: file.path })
-          }
-        },
-      )
+    if (file && !(Array.isArray(file) && file.length === 0)) {
+      const options = { maxRows: context.dataset.options?.maxRows }
+      if (key in multiAssociationLookup) {
+        const load = multiAssociationLookup[key]
+        if (!Array.isArray(file)) {
+          file = [file]
+        }
+        associations[key as keyof Associations] = await load(file, options)
```

Comment:
Didn't include a `catch` since `location` is not obvious. Might just need to silently swallow and hope the error shows up when loading the `coordsystem.json` file on its own?, e.g.,

```suggestion
        associations[key as keyof Associations] = await load(file, options).catch((e: any) => {})
```